### PR TITLE
Fix thread registry beeing removed too early

### PR DESCRIPTION
**Pull request by jvolmer**
The global registry includes a list of all thread registries. A thread registry should only be deleted from this list after it is not used any more, aka when the corresponding thread was destroyed AND all promises inside the thread registry are marked for deletion. Before, the latter condition was ignored.

### CODE REVIEW

File: arangod/AsyncRegistryServer/Feature.h [hunk 1]
```diff
@@ -43,10 +43,12 @@ class Feature final : public application_features::ApplicationFeature {
                                                  name()},
         metrics{create_metrics(
             server.template getFeature<arangodb::metrics::MetricsFeature>())} {
-    coroutine_registry.set_metrics(metrics);
+    registry.set_metrics(metrics);
     // startsAfter<Bla, Server>();
   }
 
+  ~Feature() { registry.set_metrics(std::make_shared<Metrics>()); }
```
Comment (jbajic) [inline arangod/AsyncRegistryServer/Feature.h:50]: Why do we add new metrics to the registry in dtor?

**Review by jbajic — APPROVED** [review]
Left just a question, otherwise LGTM

File: arangod/AsyncRegistryServer/Feature.h (hunk 1 above)
Comment (maierlars) [reply arangod/AsyncRegistryServer/Feature.h:50]: We replace it with an default constructed version of the metrics.

Comment (maierlars) [reply arangod/AsyncRegistryServer/Feature.h:50]: I'm also not entirely sure if we need this 🤔

**Comment by maierlars** [comment]
only ui tests are red.

**Pull request merged**